src/python/bot/tasks/corpus_pruning_task.py

```diff
@@ -682,21 +686,26 @@ def do_corpus_pruning(context, last_execution_failed, revision):
 
   logs.log('Finished.')
 
-  result = CorpusPruningResult(
-      coverage_info=coverage_info,
-      crashes=list(crashes.values()),
-      fuzzer_binary_name=fuzzer_binary_name,
-      revision=environment.get_value('APP_REVISION'))
-
   sources = ','.join([
       fuzzer.fuzz_target.project_qualified_name()
       for fuzzer in context.cross_pollinate_fuzzers
   ])
 
-  record_cross_pollination_stats(
-      pruner_stats, pollinator_stats, project_qualified_name, sources,
-      context.tag, pre_pollination_corpus_size, minimized_corpus_size_units,
-      context.cross_pollination_method)
+  if not pruner_stats or not pollinator_stats:
+    cross_pollination_stats = None
+  else:
+    cross_pollination_stats = CrossPollinationStats(
+        project_qualified_name, context.cross_pollination_method, sources,
+        context.tag, initial_corpus_size, pre_pollination_corpus_size,
+        pruner_stats['edge_coverage'], pollinator_stats['edge_coverage'],
+        pruner_stats['feature_coverage'], pollinator_stats['feature_coverage'])
+
+  result = CorpusPruningResult(
```

Comment:
nit: no need for result variable now. can just return the CorpusPruningResult directly.